hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java

```diff
@@ -1423,9 +1420,7 @@ private static boolean format(Configuration conf, boolean force,
       LOG.warn("Encountered exception during format", ioe);
       throw ioe;
     } finally {
-      if (fsImage != null) {
-        fsImage.close();
-      }
+      fsImage.close();
```

Comment:
Why should we delete this condition? @zzccctv @tasanuma

Reply:
@slfan1989 Since the `fsImage` variable is initialized with `new FSImage`, it can't be null, and IntelliJ has issued a warning that `fsImage != null is always true`. Generally, even in such situations, performing a null check is considered good practice as future code changes could potentially cause the variable to become null. However, I thought this is unlikely in this case.

Reply:
@tasanuma Thanks for the explanation! It makes sense to me. Thanks for helping review the code!

Reply:
Happy New Year, Everybody!

Leave some minor opinions, we can use the try-with-resources.
there are still some invalid variables that cause some warnings in the test report, If necessary, I can optimize it here. cc @tasanuma